packages/fdc3-agent-proxy/src/listeners/UserChannelContextListener.ts

```diff
@@ -12,5 +12,5 @@ export interface UserChannelContextListener extends Listener, RegisterableListen
    * call it's handler with the latest piece of relevant channel state and start responding to
    * events on the new channelId.
    */
-  changeChannel(ch: Channel | null): Promise<void>;
+  changeChannel(): Promise<void>;
```

Comment:
Haven't you removed the implementation of this function and all calls to it?

If so this entire interface is probably defunct/vestigal.

Reply:
Ah you moved `changeChannel()`  to `UnsubscribingDefaultContextListener, I thought you'd just dropped it. Terrible name BTW... particularly as all the context listeners already have unsubscribe fns... shame UserChannelContextListener is taken. `UserChannelContextListenerImpl` ? `ChannelHoppingContextListener`?